### PR TITLE
Float class (backend)

### PR DESCRIPTION
**Pull request by saveriomiroddi**
Backend implementation of the Float class, without the frontend (parsing). At this stage, Floats are instantiated via `String#to_f`.

Two related additions are made to the codebase:

- the `Numeric` interface, common to `Float` and `Integer`; currently it's only a lightweight, convenient, interface, not a full/visible Goby class
- the static method `toBooleanObject()`, which replaces the boilerplate for converting Go boolean values to a Goby boolean Object

This PR closes the backend side of #379.

### CODE REVIEW

**Comment by saveriomiroddi** [comment]
Reference: Ruby strategy for cross-data type operations:

    /*
     * call-seq:
     *   float + other  ->  float
     *
     * Returns a new Float which is the sum of +float+ and +other+.
     */

    static VALUE
    flo_plus(VALUE x, VALUE y)
    {
        if (RB_TYPE_P(y, T_FIXNUM)) {
      return DBL2NUM(RFLOAT_VALUE(x) + (double)FIX2LONG(y));
        }
        else if (RB_TYPE_P(y, T_BIGNUM)) {
      return DBL2NUM(RFLOAT_VALUE(x) + rb_big2dbl(y));
        }
        else if (RB_TYPE_P(y, T_FLOAT)) {
      return DBL2NUM(RFLOAT_VALUE(x) + RFLOAT_VALUE(y));
        }
        else {
      return rb_num_coerce_bin(x, y, '+');
        }
    }

And (see https://silverhammermba.github.io/emberb/c/):

    There are a host of “NUM” macros that try to be more duck-typish about things. These will convert their C types to whatever Ruby Numeric subclass seems appropriate:
    
    ...
    DBL2NUM() for double

**Comment by saveriomiroddi** [comment]
@st0012 should the equivalent of `Integer#to_intNN` be implemented for Float? For example, `Float#to_float32`? I don't know this design/concept, since there's nothing similar in Ruby.

**Comment by st0012** [comment]
@saveriomiroddi The purpose for those methods is for calling Go functions. For current Integer class, those methods just change integer object's flag, and those flags only will be used when the integer is passed as Go function's parameters.

But I think I'll change the implementation of this by letting methods like `to_int8` return a `GoObject` that really contains the `int8` type integer. Because mutation is bad.

**Comment by saveriomiroddi** [comment]
@st0012 ready for review.

File: vm/integer_test.go [hunk 1]
```diff
@@ -22,32 +22,39 @@ func TestIntegerClassSuperclass(t *testing.T) {
...
-		{`25 / 5`, 5},
-		{`25 > 5`, true},
-		{`4 > 6`, false},
-		{`-5 < -4`, true},
-		{`100 < 81`, false},
-		{`5 ** 4`, 625},
-		{`25 / 5`, 5},
-		{`1 / 1 + 1`, 2},
-		{`0 / (1 + 1000)`, 0},
-		{`5 ** (3 * 2) + 21`, 15646},
```
Comment (st0012) [inline vm/integer_test.go]: @saveriomiroddi I want to keep these combined arithmetic operations test cases. Can you create another test function for these kind of tests?

**Comment by st0012** [comment]
BTW thanks for reorganizing our integer tests at the same time 👍

File: vm/integer_test.go (hunk 1 above)
Comment (saveriomiroddi) [reply vm/integer_test.go]: :+1: Restored!

Comment (st0012) [reply vm/integer_test.go]: Nice 😎

**Pull request merged**